src/platform/linux.rs

```diff
@@ -824,56 +862,107 @@ mod desktop {
                 "ps -u {} -f | grep 'Xorg' | grep -v 'grep'",
                 &self.uid
             )) {
+                log::debug!("extracting xauth file for Xorg running with uid: {}", self.uid);
                 for line in output.lines() {
                     let mut auth_found = false;
-                    for v in line.split_whitespace() {
-                        if v == "-auth" {
-                            auth_found = true;
-                        } else if auth_found {
-                            if std::path::Path::new(v).is_absolute() {
-                                self.xauth = v.to_string();
-                            } else {
-                                if let Some(pid) = line.split_whitespace().nth(1) {
-                                    let home_dir = get_env_from_pid("HOME", pid);
-                                    if home_dir.is_empty() {
-                                        self.xauth = format!("/home/{}/{}", self.username, v);
+
+                    if let Some(pid) = line.split_whitespace().nth(1) {
```

Comment:
Why add one more `if` here?

Reply:
Thanks for review, that's a left over for Xauth file fallback. Cann t test this here so this step will be  removed again with next PR.